src/hashstore/hashstore_factory.py

```diff
@@ -0,0 +1,35 @@
+"""Core module for HashStore Factory"""
+from hashstore.filehashstore.filehashstore import FileHashStore
+
+
+class HashStoreFactory:
+    """A factory class for creating `HashStore`-like objects (classes
+    that implement the 'hashstore' abstract methods)
+
+    This factory class provides a method to retrieve a `hashstore` object
+    based on the specified store type (ex. "filehashstore"). It supports the
+    creation of different types of hash stores by mapping store types to
+    specific implementations.
+    """
+
+    def __init__(self):
+        """Initialize the HashStoreFactory with default config values"""
+        # TODO: Add logging
+
+    def get_hashstore(self, hashstore_type):
+        """Get a `HashStore`-like object based on the specified store type.
+
+        Args:
+            hashstore_type (str): The type of `HashStore` to retrieve.
+
+        Returns:
+            HashStore: A hash store object based on the given store type.
+
+        Raises:
+            ValueError: If the given store_type is not supported.
+        """
+        hashstore_type.lower()
+        if hashstore_type == "filehashstore":
+            return FileHashStore()
```

Comment:
Ideally, the factory class would not have a reference to FileHashStore or other implementations, but would only reference the HashStore interface.

It would also instantiate the FileHashStore (or whatever) from string values either passed in, or retrieved from config (like using class.forName() in Java). 

Can we do [something like this](https://stackoverflow.com/questions/4821104/dynamic-instantiation-from-string-name-of-a-class-in-dynamically-imported-module) here, instead?
```
module = __import__(module_name)
class_ = getattr(module, class_name)
instance = class_()
```

Reply:
Understood and I have made the update - however, the `__import__` class itself dissuades us from calling it and instead refers us to `importlib.import_module()` which is what I have refactored the static method to.
- `__import__`: "Import a module. Because this function is meant for use by the Python interpreter and not for general use, it is better to use importlib.import_module() to programmatically import a module."

Here's how to get the suggested above code block to execute as expected for reference (looked simple at first...):
```
module = __import__(module_name, fromlist=[class_name])
components = module_name.split('.')
for comp in components[1:]:
    module = getattr(module, comp)
hashstore_class = getattr(module, class_name)
return hashstore_class()
```
- It's not as simple as well which is why I opted to use `importlib` (in addition to `__import__`'s recommendation):
```
imported_module = importlib.import_module(module_name)
hashstore_class = getattr(imported_module, class_name)
```